types/styled-system/index.d.ts

```diff
@@ -0,0 +1,583 @@
+//                 Ben McCormick   <https://github.com/phobon>
+//                 Justin Bennett  <https://github.com/zephraph>
+//                 Christopher Pappas <https://github.com/damassi>
+//                 Eloy Durán      <https://github.com/alloy>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+// TypeScript Version: 2.1
+
+/**
+ * Core
+ */
+
+type GlobalStyleValues = "inherit" | "initial" | "unset";
+
+interface BaseTheme {
+    breakpoints?: number[];
+    space?: number[];
+    fontSizes?: number[];
+    colors?: {
+        [name: string]: string;
+    };
+}
+
+/**
+ * Space
+ */
+
+type ResponsiveValue<T> = T | Array<T | null>;
+
+type SpaceValue = number | string;
+type ResponsiveSpaceValue = ResponsiveValue<SpaceValue>;
+
+interface SpaceProps {
+    m?: ResponsiveSpaceValue;
+    mt?: ResponsiveSpaceValue;
+    mr?: ResponsiveSpaceValue;
+    mb?: ResponsiveSpaceValue;
+    ml?: ResponsiveSpaceValue;
+    mx?: ResponsiveSpaceValue;
+    my?: ResponsiveSpaceValue;
+    p?: ResponsiveSpaceValue;
+    pt?: ResponsiveSpaceValue;
+    pr?: ResponsiveSpaceValue;
+    pb?: ResponsiveSpaceValue;
+    pl?: ResponsiveSpaceValue;
+    px?: ResponsiveSpaceValue;
+    py?: ResponsiveSpaceValue;
+}
+
+declare function space(...args: any[]): any;
+
+/**
+ * Width
+ */
+
+type WidthValue = number | string;
+type ResponsiveWidthValue = ResponsiveValue<WidthValue>;
+
+interface WidthProps {
+    width?: ResponsiveWidthValue;
+}
+
+interface MinWidthProps {
+    minWidth?: ResponsiveWidthValue;
+}
+
+interface MaxWidthProps {
+    maxWidth?: ResponsiveWidthValue;
+}
+
+declare function width(...args: any[]): any;
+declare function minWidth(...args: any[]): any;
+declare function maxWidth(...args: any[]): any;
+
+/**
+ * Height
+ */
+
+type HeightValue = number | string;
+type ResponsiveHeightValue = ResponsiveValue<HeightValue>;
+
+interface HeightProps {
+    height?: ResponsiveHeightValue;
+}
+
+interface MinHeightProps {
+    minHeight?: ResponsiveHeightValue;
+}
+
+interface MaxHeightProps {
+    maxHeight?: ResponsiveHeightValue;
+}
+
+declare function height(...args: any[]): any;
+declare function minHeight(...args: any[]): any;
+declare function maxHeight(...args: any[]): any;
+
+/**
+ * Font Size
+ */
+
+type FontSizeValue = number | string;
+type ResponsiveFontSizeValue = ResponsiveValue<FontSizeValue>;
+
+interface FontSizeProps {
+    fontSize?: ResponsiveFontSizeValue;
+}
+
+declare function fontSize(...args: any[]): any;
+
+/**
+ * Color
+ */
+type ColorValue = string;
+type ResponsiveColorValue = ResponsiveValue<ColorValue>;
+
+declare function color(...args: any[]): any;
+
+/**
+ * Typography
+ */
+interface FontFamilyProps {
+    fontFamily?: string;
+}
+declare function fontFamily(...args: any[]): any;
+
+type TextAlignValue =
+    | "left"
+    | "right"
+    | "center"
+    | "justify"
+    | "justify-all"
+    | "start"
+    | "end"
+    | "match-parent";
```

Comment:
How about using https://github.com/frenic/csstype for css values?

Reply:
Oh interesting, I didn’t know you could reference external dependencies. Great! 👌 

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/fe3fcd77c952fdc23670f0a1c66431384c10862c/types/react/package.json#L4

Reply:
Awesome suggestion! I'll dig into it.